examples/profile.js

```diff
@@ -23,28 +23,71 @@ app.MainController = function($http, $scope) {
 
   $http.get('data/profile.json').then(angular.bind(this, function(resp) {
     this['profileData'] = resp.data['profile'];
+    this['profilePoisData'] = [
+      {sort: 1, dist: 1000, title: 'First POI', id: 12345},
+      {sort: 2, dist: 3000, title: 'Second POI', id: 12346}
+    ];
   }));
 
+
   /**
-   * @param {Object} item
-   * @return {number}
+   * Factory for creating simple getter functions for extractors.
+   * If the value is in a child property, the opt_childKey must be defined.
+   * The type parameter is used by closure to type the returned function.
+   * @param {T} type An object of the expected result type.
+   * @param {string} key Key used for retrieving the value.
+   * @param {string=} opt_childKey Key of a child object.
+   * @template T
+   * @return {function(Object): T}
    */
-  var z = function(item) {
-    return item['values']['mnt'];
+  var typedFunctionsFactory = function(type, key, opt_childKey) {
```

Comment:
Unfortunately, in my case it is not possible and closure complains rightfully about it:
Line 110, E:0133: Unused local variable: typedFunctionsFactory.
The `typedFunctionsFactory` must be defined before its use in the `elevationExtractor` struct.